### ISSUE TITLE
if timeout.tasks is reached pipeline status.reason should be PipelineRunTimeout

### ISSUE THREAD

**Issue by flokain** [tektoncd/pipeline#7573]
# Expected Behavior
if the tasks of a pipelineRun time out because of the timeout.tasks parameter, the reason for the PipelinRun satus should be `PipelineRunTimeout`.

# Actual Behavior
status reason is `Failed`. 

# Steps to Reproduce the Problem

1. have pipeline that has a sleep for ever task
2. configure a pipelinrun and set timeout.tasks = "1m"
4. inspect pipelinerun status = false, reason=failed.

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

```
Client Version: v1.25.3
Kustomize Version: v4.5.7
Server Version: v1.25.16-eks-8cb36c9
```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

```
Client version: 0.31.2
Chains version: v0.16.0
Pipeline version: v0.47.5
Triggers version: v0.24.0
Dashboard version: v0.35.0

```

**Comment by vdemeester** [comment]
I do think it make sense (if any Task fails due to Timeout, to mark the PipelineRun has failed due to Timeout).
cc @tektoncd/core-maintainers

**Comment by JeromeJu** [comment]
> I do think it make sense (if any Task fails due to Timeout, to mark the PipelineRun has failed due to Timeout). cc @tektoncd/core-maintainers

Yes, this would be similar to what TaskRun has for timeout behaviour.

And just double checking if there is gonna be any concern with the change cc @AlanGreene 🤔

**Comment by AlanGreene** [comment]
No concerns from the Dashboard's perspective
